### PR TITLE
Update solvable_files.sh to allow using birthday

### PR DESCRIPTION
**Pull request by platima**
Stats file for birthday, else change date, allowing these values to be used instead of just the current time

### CODE REVIEW

File: solvable_files.sh [hunk 1]
```diff
@@ -106,6 +112,7 @@ function correct_mtime() {
 
 	if [ "$mtime_in_db" == "" ]
 	then
+		echo "No mtime in database. File not indexed. Skipping $filepath"
```
Comment (artonge) [inline solvable_files.sh]: Can you make it conditional with verbose mode ?

File: solvable_files.sh [hunk 2]
```diff
@@ -115,15 +122,37 @@ function correct_mtime() {
...
 	then
-		touch -c "$filepath"
-		if [ "$scan_action" = "scan" ]
+		if [ "$use_birthday" == "use_birthday" ]
 		then
+			newdate=$(stat -c "%w" "$filepath")
+
+			if [ "$newdate" == "-" ]
+			then
+				echo "$filepath has no birthday. Using change date."
```
Comment (artonge) [inline solvable_files.sh]: Same, verbose mode

File: solvable_files.sh [hunk 3]
```diff
@@ -115,15 +122,37 @@ function correct_mtime() {
...
+		fi
+
+		if [ "$verbose" == "verbose" ]
+		then
+			echo mtime for \"$filepath\" updated to \"$(stat -c "%y" "$filepath")\"
+		fi
+
+		if [ "$scan_action" == "scan" ]
+		then
+			if [ ! -e "./occ" ]; then echo "Sorry please run this from the directory containing the 'occ' script"; exit; fi
```
Comment (artonge) [inline solvable_files.sh]:
```suggestion
			if [ ! -e "./occ" ]
			then
				echo "Please run this from the directory containing the 'occ' script"
				exit 1
			fi
```

File: solvable_files.sh [hunk 4]
```diff
@@ -115,15 +122,37 @@ function correct_mtime() {
...
 			sudo -u "$(stat -c '%U' ./occ)" php ./occ files:scan --quiet --path="$relative_filepath"
 		fi
+	elif [ "$action" == "list" ] && [ -e "$filepath" ]
+	then
+		echo -n Would update \"$filepath\" to\ 
+		if [ $use_birthday == "use_birthday" ]; then echo birthday; else echo today; fi
```
Comment (artonge) [inline solvable_files.sh]:
```suggestion
		if [ $use_birthday == "use_birthday" ]
		then
			echo birthday
		else
			echo today
		fi
```

**Review by artonge — CHANGES_REQUESTED** [review]

File: solvable_files.sh (hunk 2 above)
Comment (platima) [reply solvable_files.sh]: Sorted with sugar on top

**Review by artonge — APPROVED** [review]

**Comment by artonge** [comment]
Thanks for that @platima  :)

**Comment by platima** [comment]
All good mate. Was a painful client bug so just trying to smooth recovery!

**Pull request merged**